vendors/circom/benchmark/circom_benchmark.cc

```diff
@@ -91,20 +91,16 @@ int RealMain(int argc, char** argv) {
       witness_loader.Set("in", Uint8ToBitVector<F>(in));
       witness_loader.Load();
 
-      const zk::r1cs::ConstraintMatrices<F>& constraint_matrices =
-          tachyon_runner->constraint_matrices();
-
-      domain = Domain::Create(constraint_matrices.num_constraints +
-                              constraint_matrices.num_instance_variables);
+      domain = Domain::Create(tachyon_runner->GetDomainSize());
 
       full_assignments = base::CreateVector(
-          constraint_matrices.num_instance_variables +
-              constraint_matrices.num_witness_variables,
+          tachyon_runner->GetNumInstanceVariables() +
+              tachyon_runner->GetNumWitnessVariables(),
           [&witness_loader](size_t i) { return witness_loader.Get(i); });
 
       public_inputs =
           absl::MakeConstSpan(full_assignments)
-              .subspan(1, constraint_matrices.num_instance_variables - 1);
+              .subspan(1, tachyon_runner->GetNumInstanceVariables() - 1);
```

Comment:
Could you set `tachyon_runner->GetNumInstanceVariables()` to a temp variable to prevent running the same function twice(similar to `prover_main`)?
```suggestion
      size_t num_instance_variables =  tachyon_runner->GetNumInstanceVariables();
      full_assignments = base::CreateVector(num_instance_variables +
              tachyon_runner->GetNumWitnessVariables(),
          [&witness_loader](size_t i) { return witness_loader.Get(i); });

     public_inputs =
          absl::MakeConstSpan(full_assignments)
              .subspan(1, num_instance_variables - 1);